equinox-shipping/Domain/FinalizationTransaction.fs

```diff
@@ -0,0 +1,100 @@
+module FinalizationTransaction
+
+open Types
+open FSharp.UMX
+
+let [<Literal>] Category = "FinalizationTransaction"
+
+type State =
+    | Initial
+    | Running   of RunningState
+    | Completed of success : bool
+and RunningState =
+    | Assigning of shipmentIds : string<shipmentId>[]
+    | Assigned  of shipmentIds : string<shipmentId>[]
+    | Reverting of shipmentIds : string<shipmentId>[]
+
+type FinalizationTransactionState = { container : string<containerId> option; state : State }
+
+module Events =
+
+    let streamName (transactionId : string<transactionId>) = FsCodec.StreamName.create Category (UMX.untag transactionId)
+
+    type Event =
+        | FinalizationRequested of {| containerId: string; shipmentIds : string[] |}
+        | AssignmentCompleted   of {| shipmentIds : string[] |}
+        | RevertRequested       of {| shipmentIds : string[] |}
+        | Completed
+        interface TypeShape.UnionContract.IUnionContract
+
+    let codec = FsCodec.NewtonsoftJson.Codec.Create<Event>()
+
+type Action =
+    | AssignShipments   of containerId : string<containerId> * shipmentIds : string<shipmentId>[]
+    | FinalizeContainer of containerId : string<containerId>
+    // Reverts the assignment of the container for the shipments provided.
+    | RevertAssignment  of shipmentIds : string<shipmentId>[]
+    | Finish            of success : bool
+
+module Fold =
+
+    type State = FinalizationTransactionState
```

Comment:
should not shadow name on L8 (rename that one and/or nest it in a `module Types`)